lib/services/health_check_config.go

```diff
@@ -0,0 +1,136 @@
+	DeleteHealthCheckConfig(ctx context.Context, name string) error
+}
+
+// ValidateHealthCheckConfig validates the given health check config.
+func ValidateHealthCheckConfig(s *healthcheckconfigv1.HealthCheckConfig) error {
+	switch {
+	case s == nil:
+		return trace.BadParameter("object must not be nil")
+	case s.Version != types.V1:
+		return trace.BadParameter("only version %q is supported, got %q", types.V1, s.Version)
+	case s.Kind != types.KindHealthCheckConfig:
+		return trace.BadParameter("kind must be %q, got %q", types.KindHealthCheckConfig, s.Kind)
+	case s.Metadata == nil:
+		return trace.BadParameter("metadata is missing")
+	case s.Metadata.Name == "":
+		return trace.BadParameter("metadata.name is missing")
+	case s.Spec == nil:
+		return trace.BadParameter("spec is missing")
+	case s.Spec.Match == nil:
+		return trace.BadParameter("spec.match is missing")
+	case len(s.Spec.Match.DbLabels) == 0 && len(s.Spec.Match.DbLabelsExpression) == 0:
+		return trace.BadParameter("at least one of spec.match.db_labels or spec.match.db_labels_expression must be set")
+	}
+
+	for _, label := range s.Spec.Match.DbLabels {
+		if err := validateLabel(label); err != nil {
+			return trace.BadParameter("invalid spec.db_labels: %v", err)
+		}
+	}
+	if expr := s.Spec.Match.DbLabelsExpression; len(expr) > 0 {
+		if _, err := parseLabelExpression(expr); err != nil {
+			return trace.BadParameter("invalid spec.db_labels_expression: %v", err)
+		}
+	}
+
+	timeout := s.Spec.Timeout.AsDuration()
+	switch {
+	case timeout == 0:
+		timeout = defaults.HealthCheckTimeout
+	case timeout < constants.MinHealthCheckTimeout:
+		return trace.BadParameter("spec.timeout must be at least %s", constants.MinHealthCheckTimeout)
+	}
+
+	interval := s.Spec.Interval.AsDuration()
+	switch {
+	case interval == 0:
+		interval = defaults.HealthCheckInterval
+	case interval < constants.MinHealthCheckInterval:
+		return trace.BadParameter("spec.interval must be at least %s", constants.MinHealthCheckInterval)
+	case interval > constants.MaxHealthCheckInterval:
+		return trace.BadParameter("spec.interval must not be greater than %s", constants.MaxHealthCheckInterval)
+	}
+
+	if timeout > interval {
+		if s.Spec.Timeout.AsDuration() == 0 {
+			return trace.BadParameter("spec.interval (%s) must not be less than the default timeout (%s)", interval, defaults.HealthCheckTimeout)
+		}
+		if s.Spec.Interval.AsDuration() == 0 {
+			return trace.BadParameter("spec.timeout (%s) must not be greater than the default interval (%s)", timeout, defaults.HealthCheckInterval)
+		}
+		return trace.BadParameter("spec.timeout (%s) must not be greater than spec.interval (%s)", timeout, interval)
+	}
+	return nil
+}
+
+func validateLabel(label *labelv1.Label) error {
+	if label.Name == types.Wildcard {
+		if len(label.Values) != 1 || label.Values[0] != types.Wildcard {
+			return trace.BadParameter("selector %s:<val> is not supported", types.Wildcard)
```

Comment:
nit: it might be helpful to include actual values here instead of just `<val>`, and say what _is_ supported instead (`*: *`).